PerformanceTest_DynamicPressureAnalysis_CustomTemplate.py

```diff
@@ -0,0 +1,567 @@
+                     dorsalInstep, dorsalInstepLat, dorsalInstepMed, 
+                     plantarMat, plantarToe, plantarToeLat, plantarToeMed,
+                     plantarForefoot, plantarForefootLat, plantarForefootMed,
+                     plantarMidfoot, plantarMidfootLat, plantarMidfootMed,
+                     plantarHeel, plantarHeelLat, plantarHeelMed, plantarLateral, plantarMedial, RForce, RHS, RTO,
+                     config, movement, subj, dat)
+    
+    return(result)
+
+
+
+#############################################################################################################################################################
+
+# Read in files
+# only read .asc files for this work
+fPath = 'C:/Users/Kate.Harrison/Boa Technology Inc/PFL Team - General/Testing Segments/AgilityPerformanceData/AS_Trail_BothDialsOpen_Mech_Jan2023/Xsensor/'
+fileExt = r".csv"
+entries = [fName for fName in os.listdir(fPath) if fName.endswith(fileExt)]
+
+
+
+
+badFileList = []
+
+for fName in entries:
+    
+    config = []
+    subject = []
+    ct = []
+    movement = []
+
+    toePmidstance = []
+    toeAreamidstance = []
+    ffAreaLate = []
+    ffPLate = []
+    heelAreaLate = []
+    heelPLate = []
+    maxmaxToes = []
+
+    latPmidstance = []
+    latAreamidstance = []
+    latPLate = []
+    latAreaLate = []
+    medPmidstance = []
+    medAreamidstance = []
+    medPLate = []
+    medAreaLate = []
+    
+    latPropMid = []
+    medPropMid = []
+    dorsalVar = []
+
+    try: 
+        fName = entries[9]
+        subName = fName.split(sep = "_")[0]
+        ConfigTmp = fName.split(sep="_")[1]
+        moveTmp = fName.split(sep = "_")[2].split(sep = '.')[0].lower()
+        
+        # Make sure the files are named FirstLast_Config_Movement_Trial# - The "if" statement won't work if there isn't a trial number next to the movement
+        if ('skate' in moveTmp) or ('cmj' in moveTmp) or ('run' in moveTmp) or ('walk' in moveTmp):
+            #dat = pd.read_csv(fPath+fName, sep=',', skiprows = 1, header = 'infer')
+        
+            tmpDat = createTSmat(fName)
+            tmpDat.plotAvgPressure()
+            
+            answer = True # if data check is off. 
+            if data_check == 1:
+                plt.figure()
+                plt.plot(tmpDat.RForce, label = 'Right Foot Total Force')
+                for i in range(len(tmpDat.RHS)):
+    
+                    plt.axvspan(tmpDat.RHS[i], tmpDat.RTO[i], color = 'lightgray', alpha = 0.5)
+                answer = messagebox.askyesno("Question","Is data clean?")
+            
+            
+            
+            if answer == False:
+                plt.close('all')
+                print('Adding file to bad file list')
+                badFileList.append(fName)
+        
+            if answer == True:
+                plt.close('all')
+                print('Estimating point estimates')
+                tmpDat.plotAvgPressure()
```

Comment:
Probably want to remove this plotting for the final version. I was getting slightly confused when the previous trial was plotting during the debug plot.